src/modules/str/libkvistr.cpp

```diff
@@ -1721,41 +1721,109 @@ static bool str_kvs_fnc_split(KviKvsModuleFunctionCall * c)
 	KVSM_PARAMETERS_END(c)
 
 	if(c->params()->count() < 4)
-	    iMaxItems = -1;
+		iMaxItems = -1;
 
 	KviKvsArray * a = new KviKvsArray();
 	c->returnValue()->setArray(a);
 
-	if(iMaxItems == 0)
-		return true;
-	if(iMaxItems == 1)
+	if(szSep.isEmpty())
 	{
-		a->append(new KviKvsVariant{szStr});
+		a->set(0, new KviKvsVariant(szStr));
 		return true;
 	}
 
-	bool bWild = szFla.contains('w', Qt::CaseInsensitive);
-	bool bContainsR = szFla.contains('r', Qt::CaseInsensitive);
+	if(iMaxItems == 0)
```

Comment:
Why did you move this condition down?

Reply:
IceN9ne's rewrote  $str.split() in 2017 making use of QStringRef, that unfortunately doesn't exists anymore in Qt6. 
I reverted a part of this commit: https://github.com/kvirc/KVIrc/commit/9e8a9d9e0d38993e145dddab64b42114efc4d3f6#diff-2c8a6d9d89531ab98f04a8dce7a8d6670d4661adac0b78edd4b5a38941890befR1717